core/src/main/java/cz/xtf/core/openshift/OpenShiftBinaryManager.java

```diff
@@ -56,9 +59,13 @@ private OpenShiftBinary getBinary(String token, String username, String password
         } else {
             // If we are using an existing kubeconfig (or a default kubeconfig), we copy the original kubeconfig
             if (StringUtils.isNotEmpty(kubeconfig)) {
-                // We copy the specified kubeconfig
+                // flatten kubeconfig in case it contains certs/keys
                 try {
-                    Files.copy(Paths.get(kubeconfig), Paths.get(ocConfigPath), StandardCopyOption.REPLACE_EXISTING);
+                    Files.deleteIfExists(Paths.get(ocConfigPath));
+                    Files.write(Paths.get(ocConfigPath),
+                            Arrays.asList(new OpenShiftBinary(OpenShifts.getBinaryPath(), null)
+                                    .execute("config", "view", "--kubeconfig", kubeconfig, "--flatten")),
+                            StandardCharsets.UTF_8, StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW);
```

Comment:
```
Files.deleteIfExists(Paths.get(ocConfigPath));
```
can be removed as well as `StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW` options. Default behaviour seems enough

> If no options are present then this method works as if the [CREATE](https://docs.oracle.com/javase/7/docs/api/java/nio/file/StandardOpenOption.html#CREATE), [TRUNCATE_EXISTING](https://docs.oracle.com/javase/7/docs/api/java/nio/file/StandardOpenOption.html#TRUNCATE_EXISTING), and [WRITE](https://docs.oracle.com/javase/7/docs/api/java/nio/file/StandardOpenOption.html#WRITE) options are present.